hs/src/Reach/Backend/JS.hs

```diff
@@ -456,6 +456,19 @@ jsExpr = \case
             [ ("amt", zero')
             , ("tok", tok')
             ]
+  DLE_CanReceive _ addr tok -> do
+    (ctxt_mode <$> ask) >>= \case
+      JM_Backend -> do
+        -- TODO - I'm not really sure what to do here, but I imagine I should do something similar to the simulate case.
+        --addr' <- jsArg addr
+        --tok' <- jsArg tok
+        --return $ "await" <+> jsApply (addr' <> ".tokenAccepted") [tok']
+        return "false"
+      JM_View -> impossible "view canReceive"
+      JM_Simulate -> do
+        addr' <- jsArg addr
+        tok' <- jsArg tok
+        return $ "await" <+> jsApply (addr' <> ".tokenAccepted") [tok']
```

Comment:
A few things...
- Simulation case --- This needs to log the address as one of the addresses that will be included in the txn. That may require a new simulation function in the JS contract interface (which will require an ABI version increment)
- Backend case --- You can't actually call `address.tokenAccepted`, because this code may be evaluated at a different time than the consensus code, after the user has opted-out and can no longer receive it. Thus, you'll need to log the result (but using `DLE_EmitLog` in the code generation in `Eval`)

Reply:
Can you tell me a little about this bit of code?  My current belief (aka best guess) is that JM_Backend is essentially for code running in local steps, JM_Simulate is for the JS to simulate consensus steps, and JM_View is for simulating views (which on Algorand must be simulated but which on ETH are actually run on a node).  Is that correct?

Reply:
`JM_Backend` is for "client-side code", which is when the client runs the consensus code locally AND for local steps. (Every consensus computation is duplicated locally.)

`JM_Simulate` is for simulating the next consensus step to compute what resources will be used. You'll need to add `addr` to the account list AND `tok` to the token list

You have to add something like this... https://github.com/reach-sh/reach-lang/blob/master/js/stdlib/ts/ALGO.ts#L1821-L1823 .... but call `recordAccount` and `recordToken` (but no fee). You'll be adding a new kind of simulation txn... https://github.com/reach-sh/reach-lang/blob/master/js/stdlib/ts/shared_impl.ts#L570

Reply:
I think I've mostly addressed things here, except I still don't know what to do for the JM_Backend case.  I don't see any other examples of JM_Backend that give me any useful hints.  It needs to return a bool... but it's a bool determined by the state of the consensus network.  So... I'm basically lost for how we can do that besides querying the network.

Reply:
You should have it just return `true`. The `EmitLog` thing will ignore it